ui/components/app/modals/visit-support-data-consent-modal/visit-support-data-consent-modal.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useCallback, useContext } from 'react';
+import { useSelector } from 'react-redux';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { selectSessionData } from '../../../../selectors/identity/authentication';
+import { getMetaMetricsId } from '../../../../selectors/selectors';
+import { openWindow } from '../../../../helpers/utils/window';
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Box,
+  ModalFooter,
+  ButtonPrimary,
+  ButtonPrimarySize,
+  ModalBody,
+  Text,
+  ButtonSecondary,
+  ButtonSecondarySize,
+} from '../../../component-library';
+import {
+  Display,
+  TextVariant,
+  BlockSize,
+} from '../../../../helpers/constants/design-system';
+import {
+  MetaMetricsContextProp,
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../../shared/constants/metametrics';
+import { MetaMetricsContext } from '../../../../contexts/metametrics';
+import { SUPPORT_LINK } from '../../../../../shared/lib/ui-utils';
+
+type VisitSupportDataConsentModalProps = {
+  onClose: () => void;
+  isOpen: boolean;
+};
+
+const VisitSupportDataConsentModal: React.FC<
+  VisitSupportDataConsentModalProps
+> = ({ isOpen, onClose }) => {
+  const version = process.env.METAMASK_VERSION;
+  const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
+  const sessionData = useSelector(selectSessionData);
+  const profileId = sessionData?.profile?.profileId;
+  const metaMetricsId = useSelector(getMetaMetricsId);
+
+  const handleClickContactSupportButton = useCallback(() => {
+    onClose();
+    const supportLinkWithUserId = `${SUPPORT_LINK}?metamask_version=${version}&metamask_profile_id=${profileId}&metamask_metametrics_id=${metaMetricsId}`;
+    trackEvent(
+      {
+        category: MetaMetricsEventCategory.Settings,
+        event: MetaMetricsEventName.SupportLinkClicked,
+        properties: {
+          url: supportLinkWithUserId,
```

Comment:
I wonder if we could make this data more useful / easier to parse on the dashboards. We could pass the version directly as a property for example, for statistics.

Reply:
Thanks for suggestions, passed in as param now 915721b0ce087a8c5afa8f76ca5d04db4114e233

Reply:
I meant something like
```javascript
// ...
properties: {
  url: supportLinkWithUserId,
  version
}
```
so PMs can see which versions redirected the most to the support page

Reply:
Oh the properties was defined before in this event, i didn't modify the logic, i'm worried it could cause inconsistency of previous events and the new ones.

Reply:
fair point, let's leave it as is then